recipes/gtsam/all/conandata.yml

```diff
@@ -1,14 +1,28 @@
 sources:
+  "4.2.0a9":
```

Comment:
@valgur I realized that the official version is `4.2a9`, should not it be better to put the same one? Am I missing something?

Reply:
I based it on the full version number in CMakeLists.txt: https://github.com/borglab/gtsam/blob/4.2a9/CMakeLists.txt#L10-L13
My reasoning was that the official release is going to be `4.2.0`, so this would be slightly more consistent and would avoid potential ambiguity when sorting the version numbers.
I don't have anything against renaming it to `4.2a9` either, though.